autorest/azure/persist_test.go

```diff
@@ -133,11 +133,13 @@ func TestSaveToken(t *testing.T) {
 
 func TestSaveTokenFailsNoPermission(t *testing.T) {
 	pathWhereWeShouldntHavePermission := "/usr/thiswontwork/atall"
+	expectedSubstring := "failed to create directory"
+
 	if runtime.GOOS == "windows" {
 		pathWhereWeShouldntHavePermission = "c:\\windows\\system32\\mytokendir\\mytoken"
+		expectedSubstring = "failed to create the temp file to write the token"
```

Comment:
This is fairly confusing to me. Your windows user account has permissions to `mkdir` on `C:\\windows\\system32\\mytokendir` but unable to write a new file inside? You didn't create the `mytokendir` as an admin user did you? Just confused as to what's going on here. As-is it's currently testing different errors on Linux/Windows.